src/typecheck/mod.rs

```diff
@@ -353,6 +352,11 @@ pub struct Envs<'a> {
     local: Environment,
 }
 
+#[derive(Clone, Debug)]
+pub enum EnvBuildError {
```

Comment:
If there's only one variant (and we don't foresee additional ones), you can also use a struct `pub struct EnvBuildError(RichTerm)`. Once again, it's nitpicking.

Reply:
It's the same in `eval::EnvBuildError` so to stay consistent, I used the same form. Also, will bee easyer to update with others errors types if needed.